src/components/forms/Combobox.tsx

```diff
@@ -790,23 +798,67 @@ export function ComboboxAsync<T = any>({
 
           return data;
         }
-      );
-    },
+      ),
     {
       staleTime: staleTime ?? Infinity,
-      refetchOnWindowFocus: false,
-      refetchOnReconnect: false,
+      enabled: enableQuery,
     }
   );
 
+  useEffect(() => {
+    if (!enableQuery) {
+      clearTimeout(enableQueryTimeOut.current);
+
+      const currentTimeout = setTimeout(() => setEnableQuery(true), 50);
+
+      enableQueryTimeOut.current = currentTimeout;
+    }
+  }, [inputOptions.value]);
+
+  useEffect(() => {
+    if (
+      enableQuery &&
+      inputOptions.value &&
+      !disableWithQueryParameter &&
+      !isEntryAvailable()
+    ) {
+      setUrl((c) => {
+        const currentUrl = new URL(c);
+
+        inputOptions.value &&
+          currentUrl.searchParams.set('with', inputOptions.value.toString());
+
+        return currentUrl.href;
+      });
+    }
+  }, [enableQuery, inputOptions.value]);
+
   useEffect(() => {
     if (data) {
       setEntries([...data]);
     }
   }, [data]);
 
   useEffect(() => {
-    return () => setEntries([]);
+    setUrl((c) => {
+      const currentUrl = new URL(c);
+
+      if (sortBy) {
+        currentUrl.searchParams.set('sort', sortBy);
+      }
+
+      currentUrl.searchParams.set('status', 'active');
+
+      currentUrl.searchParams.set('filter', '');
```

Comment:
@beganovich Not having the `filter` parameter during the initial render results in double queries. We will have one query with the `filter` parameter and one without it. However, the component itself sets the `filter` to an empty string, so to avoid double queries, I set it during the component building.